mkat_fpga_tests/test_cbf.py

```diff
@@ -451,19 +450,48 @@ def test_delay_tracking(self):
         # Choose baseline for phase comparison
         baseline_index = baseline_lookup[('m000_x', 'm000_y')]
 
-        correlation_bw = self.corr_freqs.bandwidth
         sampling_period = self.corr_freqs.sample_period
-        test_delays = [0, sampling_period, 1.5*sampling_period]
-
-        for delay in test_delays[0:1]:
-            # set delay on correlator input m000_y
-            # TODO
-            this_freq_dump = self.receiver.get_clean_dump(DUMP_TIMEOUT)
-            data = complexise(this_freq_dump['xeng_raw'][:, baseline_index, :])
-            phases = np.angle(data)
-            plt.plot(self.corr_freqs.chan_freqs, phases)
-            #import IPython; IPython.embed()
-            #plt.show()
+        test_delays = [0, sampling_period, #1.5*sampling_period,
+            3*sampling_period]
+
+        def expected_phases(plot=False):
+            expected_chan_phase = []
+            for channel in self.corr_freqs.chan_freqs:
+                phases = channel * 2 * np.pi * sampling_period
+                expected_chan_phase.append(phases)
+
+            plt.plot(self.corr_freqs.chan_freqs,
+                np.array(expected_chan_phase))
+            if plot:
+                plt.show()
+            return np.array(expected_chan_phase)
+
+        def actual_phases(plot=False):
+            actual_phases_list = []
+            for delay in test_delays:
+                # set coarse delay on correlator input m000_y
+                input_y = [s for s in self.correlator.fengine_sources
+                           if s.name == 'm000_y'][0]
+                delay_samples = int(np.floor(delay/sampling_period))
+                input_y.host.registers.coarse_delay0.write(coarse_delay=delay_samples)
+                input_y.host.registers.tl_cd0_control0.write(arm='pulse', load_immediate=1)
+
+                this_freq_dump = self.receiver.get_clean_dump(DUMP_TIMEOUT)
+                data = complexise(this_freq_dump['xeng_raw'][:, baseline_index, :])
+                phases = np.unwrap(np.angle(data))
+                actual_phases_list.append(phases)
+                plt.plot(self.corr_freqs.chan_freqs, phases)
+                if plot:
+                    plt.show()
+            return actual_phases_list
+
+        # Compare Actual and Expected phases and check if their equal
+        # upto 3 decimal places
+        self.assertAlmostEqual(np.max(np.abs(actual_phases()[1])),
```

Comment:
Why are you only testing the min/max values? We should be comparing the phases of all the channels.